spec/EmailVerificationToken.spec.js

```diff
@@ -52,6 +52,37 @@ describe('Email Verification Token Expiration: ', () => {
       });
   });
 
+  it('should send an HTML or properly escaped plain text password reset email', async () => {
+    const user = new Parse.User();
+    let sendEmailOptions;
+    const emailAdapter = {
+      sendPasswordResetEmail: (options) => {
+        sendEmailOptions = options;
+      },
+      sendVerificationEmail: async () => {},
+      sendMail: async () => {},
+    };
+
+    await reconfigureServer({
+      appName: 'specialCharacterUsernameTest',
+      publicServerURL: 'http://localhost:8378/1',
+      emailAdapter: emailAdapter,
+    });
+
+    user.setUsername('hello :)');
```

Comment:
Could you add a loop and test a bunch of different chars?

Reply:
I expanded the individual test to cover all the characters